fubectl.source

```diff
@@ -161,7 +168,7 @@ klog() {
     [ -z "$arg_pair" ] && printf "klog: no pods found. no logs can be shown.\n" && return
     local containers_out=$(echo "$arg_pair" | xargs kubectl get po -o=jsonpath='{.spec.containers[*].name} {.spec.initContainers[*].name}' -n | sed 's/ $//')
     local container_choosen=$(echo "$containers_out" |  tr ' ' "\n" | _inline_fzf_nh)
-    kubectl logs -n ${arg_pair} -c "${container_choosen}" --tail="${line_count}" "$@"
+    _kctl_tty logs -n ${arg_pair} -c "${container_choosen}" --tail="${line_count}" "$@"
```

Comment:
Sorry I'm kinda ignorant about msys, can you please explain why `kubectl logs` (or exec or port-forward) need winpty, but `kubectl get` is OK without it?

Reply:
> Sorry I'm kinda ignorant about msys, can you please explain why `kubectl logs` (or exec or port-forward) need winpty, but `kubectl get` is OK without it?

I don't exactly know, so I can explain the real reason behind this. Smth related to fact that kubectl is natively compiled tool that works in cmd.exe properly, but it sometimes behaves weird when running via `mintty` (the msys2 terminal).

So, for example, `kubectl get` outputs fine, but when you run `kubeclt logs -f` ( -f is important ) - then then press Ctrl+C - the process is not actually interrupted, so console can't be closed properly, the following logs will appear all of a sudden.

But when running with `winpty kubectl logs -f` - Ctrl + C is correctly processed and kubectl process will be properly shutdown.

This is one of possible scenarios, sometimes you need to run via winpty sometimes not :(